engine/schema/src/main/java/com/cloud/network/dao/NetworkDaoImpl.java

```diff
@@ -419,6 +419,16 @@ public int getOtherPersistentNetworksCount(long id, String broadcastURI, boolean
         return persistentNetworks.size();
     }
 
+    @Override
+    public List<NetworkVO> getAllPersistentNetworksFromZone(long dataCenterId) {
+        Object[] guestTypes = {"Isolated", "L2"};
```

Comment:
@Pearl1594 
cannot Isolated network be persistent ?

Reply:
they can @weizhouapache hence Isolated & l2 are included in the search criteria

Reply:
@Pearl1594 
oh thanks... stupid me. I thought it contains Shared and L2 ...

why exclude Shared networks ?

Reply:
We dont have a concept of Persistent shared networks currently @weizhouapache

Reply:
@Pearl1594 ok.

Reply:
@Pearl1594 
thanks for your replies and sharing.

Now I am clear the proposal of the feature. 
I think the feature is not very useful,  In mose cases, user just needs to put the physical devices in the specified vlan, it is not needed to configure bridges/vlans on hosts. Maybe vmware distributed port group is exception, it needs to be created in advance. 

on the other hand, it is not harmful to create bridges on hosts. so I am not onject to this feature.

PS1:  Persistent Shared network makes more sense IMHO.
PS2:  Do you know any user or customer using this feature ?